util/src/main/java/io/kubernetes/client/util/authenticators/OpenIDConnectAuthenticator.java

```diff
@@ -0,0 +1,246 @@
+
+        // expired now is >= expiration AND exp is present
+        return claims.getExpirationTime() == null
+            || NumericDate.now().isOnOrAfter(claims.getExpirationTime());
+      } catch (JoseException | InvalidJwtException | MalformedClaimException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  @Override
+  public Map<String, Object> refresh(Map<String, Object> config) {
+    String issuer = (String) config.get(OIDC_ISSUER);
+    String clientId = (String) config.get(OIDC_CLIENT_ID);
+    String refreshToken = (String) config.get(OIDC_REFRESH_TOKEN);
+    String clientSecret = (String) config.get(OIDC_CLIENT_SECRET);
+    String idpCert = (String) config.get(OIDC_IDP_CERT_DATA);
+
+    SSLContext sslContext = null;
+
+    if (idpCert != null) {
+      // fist, lets get the pem
+      String pemCert = new String(Base64.getDecoder().decode(idpCert));
+
+      // next lets get a cert object
+      String alias = "doenotmatter";
+      KeyStore ks;
+
+      try {
+        ks = java.security.KeyStore.getInstance("PKCS12");
+        ks.load(null, "doesnotmatter".toCharArray());
+        ByteArrayInputStream bais = new ByteArrayInputStream(pemCert.getBytes("UTF-8"));
+        CertificateFactory cf = CertificateFactory.getInstance("X.509");
+        Collection<? extends java.security.cert.Certificate> c = cf.generateCertificates(bais);
+
+        if (c.size() > 1) {
+          int j = 0;
+          Iterator<? extends java.security.cert.Certificate> i = c.iterator();
+          while (i.hasNext()) {
+            Certificate certificate = (Certificate) i.next();
+            ks.setCertificateEntry(alias + "-" + j, certificate);
+          }
+        } else {
+          ks.setCertificateEntry(alias, c.iterator().next());
+        }
+
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance("PKIX");
+        tmf.init(ks);
+
+        sslContext = SSLContext.getInstance("TLSv1.2");
+        sslContext.init(null, tmf.getTrustManagers(), new SecureRandom());
+
+      } catch (KeyStoreException
+          | NoSuchAlgorithmException
+          | CertificateException
+          | IOException
+          | KeyManagementException e) {
+        throw new RuntimeException("Could not import idp certificate", e);
+      }
+    }
+
+    if (clientSecret == null) {
+      clientSecret = "";
+    }
+
+    StringBuilder sb = new StringBuilder();
+    sb.append(issuer);
+    if (!issuer.endsWith("/")) {
+      sb.append("/");
+    }
+    sb.append(".well-known/openid-configuration");
+
+    String wellKnownURL = sb.toString();
+
+    try {
+      URL wellKnown = new URL(wellKnownURL);
+      HttpsURLConnection https = (HttpsURLConnection) wellKnown.openConnection();
+      https.setRequestMethod("GET");
+      if (sslContext != null) {
+        https.setSSLSocketFactory(sslContext.getSocketFactory());
+      }
+      https.setUseCaches(false);
+      int code = https.getResponseCode();
+      if (code == 200) {
```

Comment:
Rather than 

```java
if (code == 200) {
   //long block of code
} else {
   // throw
}
```

It is preferable to do:

```java
if (code != 200) {
   // throw exception
}
// long block of code here
```

That keeps the flow of the code easier to read because it reduces indentation.